### PR TITLE
Runtime environment variables docs for node integration

### PR DESCRIPTION
**Pull request by arty-name**
## Changes

This PR adds a section to the node integration docs on how to handle the runtime environment variables. It is not documented anywhere, and the current behavior even caused me to [file a bug about it](https://github.com/withastro/astro/issues/7195).

Since I am filing this PR via the GitHub website I cannot provide a changeset.

## Testing

I have added no tests for the modified README file because none existed.

## Docs

This is somehow a docs PR

 /cc @withastro/maintainers-docs for feedback!

### CODE REVIEW

File: packages/integrations/node/README.md
Comment (sarah11918) [inline packages/integrations/node/README.md:155]:
``````suggestion
If an `.env` file containing environment variables is present when the build process is run, these values will be hard-coded in the output, just as when generating a static website. 

For a value to be provided at run-time, use the following variable with any placeholder value in your `.env` file:  `RUNTIME_VALUE=placeholder astro build`.  This variable will signal to Astro that the actual value will be available when the built application is run. The placeholder value will be ignored by the build process, and Astro will use the value provided at run-time. 

In the case of multiple run-time variables, store them in a seperate file (e.g. `.env.runtime`) outside of `.env`. Start the build with the following command: 
```sh
export $(cat .env.runtime) && astro build
```
``````
I've edited this to split up the content into easier to read chunks, and tried to tighten up some wording while I'm at it. Of course, when I do this, I might often cut something important or miss a nuance, so please have a read over this as a suggested model, and don't hesitate to correct so that it is actually factually accurate!  :smile:

**Review by sarah11918 — COMMENTED** [review]
Thanks for this docs contribution @arty-name! I think this is very useful information to have!  

See my take on an editing pass below to fit it in with our current docs style, and see what you think!

File: packages/integrations/node/README.md
Comment (arty-name) [reply packages/integrations/node/README.md:155]: Thank you, Sarah, I’m always happy to get an input from a tech writer and a native speaker!

This topic is unfortunately overly complex, and I see that I could not explain it clearly enough. For one, it’s not the single `RUNTIME_VALUE` which needs to be provided: during the build the developer needs to provide *every* value which will also come in during the execution. Another important part which lost its emphasis is this: "Note that these variables **must be absent** in the `.env` file." 

What do you think about this improved version?

>If an `.env` file containing environment variables is present when the build process is run, these values will be hard-coded in the output, just as when generating a static website. 

>Special handling is required for values that have to be provided at run-time. During the build, they **must be absent** from the `.env` file. But you also need to tell Astro which environment variables to expect at run-time. For that you need to provide them during the build process: `VALUE1=placeholder VALUE2=placeholder astro build`. Feel free to replace `placeholder` with any non-falsy value, it will be ignored anyway.

>Providing multiple values in this way could be cumbersome. For convenience, store them in a seperate file (e.g. `.env.runtime`). Start the build with the following command: 
```sh
export $(cat .env.runtime) && astro build
```

Comment (TheOtterlord) [inline packages/integrations/node/README.md:155]: > https://github.com/withastro/astro/pull/7378#discussion_r1233608066

After the first section, you explain how to use it with multiple variables. We wouldn't want to lead people down the wrong path before asking them to do it another way. We can probably still make this clear in the text with something like "you must provide Astro with every environment variable".

In the final section, we shouldn't need to mention it's convenience, because we're trying to teach the user the one, easy way to do something.

```suggestion
If an `.env` file containing environment variables is present when the build process is run, these values will be hard-coded in the output, just as when generating a static website.

During the build, the variable must be absent from the `.env` file, and you must provide Astro with every environment variable to expect at run-time: `VARIABLE_1=placeholder astro build`. This signals to Astro that the actual value will be available when the built application is run. The placeholder value will be ignored by the build process, and Astro will use the value provided at run-time.

In the case of multiple run-time variables, store them in a seperate file (e.g. `.env.runtime`) from `.env`. Start the build with the following command:

```sh
export $(cat .env.runtime) && astro build
```
```

Comment (TheOtterlord) [reply packages/integrations/node/README.md:155]: Sorry for the long wait. Our docs reviewing/editing has had a huge backlog and I'm picking this up for Astro Docs Maintainers!

This is looking much better! I have a few suggestions based on your revision, of which I've thrown in a new review.

File: packages/integrations/node/README.md
Comment (arty-name) [inline packages/integrations/node/README.md:157]: Thank you for your input @TheOtterlord, this looks very nice! Before merging it, though, do you think it would make sense to make this clearer? "During the build, the variable must be absent from the `.env` file" does not say which variable. Could it be "the runtime variables"

Comment (TheOtterlord) [reply packages/integrations/node/README.md:157]: Yep! That sounds like a great clarification

Comment (arty-name) [reply packages/integrations/node/README.md]: I’ve adjusted it like this. How does the PR look to you now?

**Review by TheOtterlord — APPROVED** [review]
Thanks a lot @arty-name!
This is amazing! I'll update the branch, and get this merged.

**Pull request merged**
